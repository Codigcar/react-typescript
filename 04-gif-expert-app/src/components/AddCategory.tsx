import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = (props:any) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e: any) => {
    setInputValue(e.target.value);
};

const handleSubmit = (e: any) => {
    e.preventDefault();
    if(inputValue.trim().length > 2){
      props.setCategories((categories: any) => [inputValue, ...categories,]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        onChange={handleInputChange}
        type="text"
        className="input-text"
      />
    </form>
  );
};

AddCategory.prototype = {
  props: PropTypes.func.isRequired
}