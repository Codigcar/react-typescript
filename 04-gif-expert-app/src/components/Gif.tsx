import React from 'react'
import PropTypes from 'prop-types';

export const Gif = (props:any) => {
    return (
        <div className="card">
            <img src={props.url} alt={props.title} />
            <p>{props.title}</p>
        </div>
    )
}
/* export const Gif = ({title, url}:any) => {
    return (
        <div className="card">
            <img src={url} alt="props.title" />
            <p>{title}</p>
        </div>
    )
} */

Gif.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
