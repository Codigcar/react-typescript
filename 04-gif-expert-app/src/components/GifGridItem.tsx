import React from 'react'

interface propsInterface {
    category:string;
}
type imgType = {
    id: string;
    title: string;
    images: { downsized_medium: { url: string; }; }; 
}

export const GifGridItem = (props:propsInterface) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vfV8SOvefLtVRj5pOOTcRZgepQ0zs29o';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gif = data.map((img: imgType) => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gif);
        
    }
    getGifs();
    return (
        <div>
            {props.category}
        </div>
    )
}
