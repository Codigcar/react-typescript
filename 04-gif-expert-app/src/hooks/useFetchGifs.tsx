import { useEffect } from 'react';
import { useState } from 'react';
import { getGifs } from '../helpers/getGifs';


export const useFetchGifs = (category:string) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifs(category)
            .then(gifs => {
                setTimeout(() => {
                    setState({
                        data: gifs,
                        loading: false
                    }) 
                }, 3000);
            })
    }, [category])

    return state;
}