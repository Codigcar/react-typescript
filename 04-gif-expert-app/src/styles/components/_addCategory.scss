.input-text {
    width: 100%;
    color: yellowgreen;
}