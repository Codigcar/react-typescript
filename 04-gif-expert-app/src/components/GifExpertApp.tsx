import React from "react";
import { useState } from "react";
import { AddCategory } from "./AddCategory";

export const GifExpertApp = () => {
  // const categories = ['category1','category2','category3'];
  const [categories, setCategories] = useState([
    "category1",
    "category2",
    "category3",
  ]);

  return (
    <>
      <h1>Encuentra tus gifs </h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>  
        {categories.map((categories, i) => {
          return <li key={i}>{categories}</li>;
        })}
      </ol>
    </>
  );
};
