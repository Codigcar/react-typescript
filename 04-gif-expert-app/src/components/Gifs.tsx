import React from 'react'
import { useState } from 'react';
import { useEffect } from 'react';
import { Gif } from './Gif';

interface propsInterface {
    category:string;
}
type gifType = {
    id: string;
    title: string;
    images: { downsized_medium: { url: string; }; }; 
}

export const Gifs = (props:propsInterface) => {

    const [gifs, setGifs] = useState([]);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vfV8SOvefLtVRj5pOOTcRZgepQ0zs29o';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const arrayGifs = data.map((gif: gifType) => {
            return {
                id: gif.id,
                title: gif.title,
                url: gif.images?.downsized_medium.url
            }
        })
        console.log(arrayGifs);
        setGifs(arrayGifs);
    }

    useEffect(() => {
        getGifs();
    }, [])

    return (
        <div>
            <h3>{props.category}</h3>
            <ol>
                {
                    /* gifs.map((gif:gifType) => (
                        <li key={gif.id}>{ gif.title }</li>
                    )) */
                    gifs.map((gif:gifType) => (
                        <Gif key={gif.id} {...gif} />
                    ))
                }
            </ol>
        </div>
    )
}
