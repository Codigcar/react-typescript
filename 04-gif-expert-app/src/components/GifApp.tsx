import React from "react";
import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { Gifs } from "./Gifs";

export const GifExpertApp = () => {
  // const categories = ['category1','category2','category3'];
  const [categories, setCategories] = useState(["category1"]);

  return (
    <>
      <h1>Encuentra tus gifs </h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>  
        {
            categories.map((category, i) => (
            //   return <li key={i}>{categories}</li>;
                <Gifs key={i} category={category} />
            ))
        }
      </ol>
    </>
  );
};
