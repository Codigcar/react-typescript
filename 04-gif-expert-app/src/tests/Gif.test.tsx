import { shallow } from 'enzyme';
import 'jest';
import { Gif } from '../components/Gif';
import { useFetchGifs } from '../hooks/useFetchGifs';
describe('Pruebas a <Gif />', () => {
    test('debe mostrar el componente <Gif /> renderizado', () => {
        // const {data, loading}  = useFetchGifs('goku');
        // const gif = data[0];
        const wrapper = shallow( <Gif />);
        expect(wrapper).toMatchSnapshot();
        
    })
    
})
