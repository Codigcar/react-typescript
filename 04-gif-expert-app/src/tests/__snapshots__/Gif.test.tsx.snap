// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <Gif /> debe mostrar el componente <Gif /> renderizado 1`] = `
<div
  className="card"
>
  <img
    alt="props.title"
  />
  <p />
</div>
`;
